201-blob-storage-private-endpoint/azuredeploy.json

```diff
@@ -0,0 +1,900 @@
+      "type": "string",
+      "defaultValue": "https://paolosalvatori.blob.core.windows.net/scripts",
+      "metadata": {
+        "description": "The relative path of the script to download"
+      }
+    },
+    "scriptFileName": {
+      "type": "string",
+      "defaultValue": "nslookup.sh",
+      "metadata": {
+        "description": "The name of the script to download from the URI specified by the scriptFilePath parameter."
+      }
+    },
+    "deployLogAnalytics": {
+      "type": "bool",
+      "defaultValue": true,
+      "metadata": {
+        "description": "Specifies whether to deploy Log Analytics."
+      }
+    },
+    "workspaceName": {
+      "type": "string",
+      "metadata": {
+        "description": "Name of the Log Analytics workspace"
+      }
+    },
+    "workspaceSku": {
+      "type": "string",
+      "allowedValues": [
+        "PerGB2018",
+        "Free",
+        "Standalone",
+        "PerNode",
+        "Standard",
+        "Premium"
+      ],
+      "defaultValue": "PerGB2018",
+      "metadata": {
+        "description": "Pricing tier: PerGB2018 or legacy tiers (Free, Standalone, PerNode, Standard or Premium) which are not available to all customers."
+      }
+    },
+    "privateEndpointName": {
+      "type": "string",
+      "defaultValue": "StorageAccountPrivateEndpoint",
+      "metadata": {
+        "description": "Specifies the name of the private link to Azure Database for PostgreSQL."
+      }
+    }
+  },
+  "variables": {
+    "nicName": "[concat(parameters('vmName'), 'Nic')]",
+    "nsgName": "[concat(parameters('subnetName'), 'SubnetNsg')]",
+    "publicIPAddressName": "[concat(parameters('vmName'), 'PublicIp')]",
+    "publicIPAddressType": "Dynamic",
+    "storageAccountId": "[resourceId('Microsoft.Storage/storageAccounts', parameters('storageAccountName'))]",
+    "workspaceId": "[resourceId('Microsoft.OperationalInsights/workspaces', parameters('workspaceName'))]",
+    "subnetId": "[resourceId('Microsoft.Network/virtualNetworks/subnets', parameters('virtualNetworkName'), parameters('subnetName'))]",
+    "publicIpId": "[resourceId('Microsoft.Network/publicIPAddresses', variables('publicIPAddressName'))]",
+    "nicId": "[resourceId('Microsoft.Network/networkInterfaces', variables('nicName'))]",
+    "vnetId": "[resourceId('Microsoft.Network/virtualNetworks', parameters('virtualNetworkName'))]",
+    "nsgId": "[resourceId('Microsoft.Network/networkSecurityGroups',variables('nsgName'))]",
+    "vmId": "[resourceId('Microsoft.Compute/virtualMachines', parameters('vmName'))]",
+    "linuxConfiguration": {
+      "disablePasswordAuthentication": true,
+      "ssh": {
+        "publicKeys": [
+          {
+            "path": "[concat('/home/', parameters('adminUsername'), '/.ssh/authorized_keys')]",
+            "keyData": "[parameters('adminPasswordOrKey')]"
+          }
+        ]
+      },
+      "provisionVMAgent": true
+    },
+    "publicDNSZoneForwarder": ".blob.core.windows.net",
```

Comment:
this cannot be hard-coded, you can use environment().suffixes.storage to get "core.windows.net" - concat what you need from there

Reply:
Solved! I used the suffix returned by environment().suffixes.storage as per your guidance